src/select.ts

```diff
@@ -11,17 +10,42 @@ export class NgrxSelect {
   }
 }
 
-export function Select(path?: string): any {
-  return function(target: any, name: string, descriptor: TypedPropertyDescriptor<any>): void {
+export function Select<TState = any, TValue = any>(
+  selector: Selector<TState, TValue>
+): (target: any, name: string) => void;
+export function Select<TState = any, TValue = any>(
+  selectorOrFeature?: string,
+  ...paths: string[]
+): (target: any, name: string) => void;
+export function Select<TState = any, TValue = any>(
+  selectorOrFeature: string | Selector<TState, TValue>,
```

Comment:
overload signature is not compatible with function implementation,

should be `selectorOrFeature?: string | ...`